libs/execution_context/ebpf_maps.c

```diff
@@ -939,10 +676,33 @@ ebpf_map_update_entry(
         return EBPF_INVALID_ARGUMENT;
     }
 
-    if (ebpf_map_function_tables[map->ebpf_map_definition.type].update_entry == NULL) {
+    if ((ebpf_map_function_tables[map->ebpf_map_definition.type].update_entry == NULL) ||
+        (ebpf_map_function_tables[map->ebpf_map_definition.type].find_entry == NULL)) {
         return EBPF_INVALID_ARGUMENT;
     }
-    return ebpf_map_function_tables[map->ebpf_map_definition.type].update_entry(map, key, value);
+
+    if (flags & EBPF_MAP_FLAG_HELPER) {
+        uint8_t* target = ebpf_map_function_tables[map->ebpf_map_definition.type].find_entry(map, key);
+        if (!target) {
+            ebpf_result_t return_value =
+                ebpf_map_function_tables[map->ebpf_map_definition.type].update_entry(map, key, NULL);
+            if (return_value != EBPF_SUCCESS) {
+                return return_value;
+            }
+            target = ebpf_map_function_tables[map->ebpf_map_definition.type].find_entry(map, key);
+            if (!target) {
+                return EBPF_NO_MEMORY;
+            }
+        }
+        if (_ebpf_adjust_value_pointer(map, &target) != EBPF_SUCCESS) {
+            return EBPF_INVALID_ARGUMENT;
+        }
+
+        memcpy(target, value, ebpf_map_get_effective_value_size(map));
```

Comment:
In case update means more than memcpy for any map type, I suggest replacing memcpy with an invocation into the map_type implementation.

Reply:
ok